qemu/tests/seabios.py

```diff
@@ -34,24 +35,104 @@ def boot_menu_check():
         return (len(re.findall(boot_menu_hint,
                                get_output(seabios_session))) > 1)
 
-    error.context("Start guest with sga bios")
     vm = env.get_vm(params["main_vm"])
     # Since the seabios is displayed in the beginning of guest boot,
     # booting guest here so that we can check all of sgabios/seabios
     # info, especially the correct time of sending boot menu key.
     vm.create()
+    seabios_session = vm.logsessions['seabios']
+
+    if 'machine_type_bin_check' in params:
+        login_timeout = float(params.get("login_timeout", 360))
+        qemu_binary = utils_misc.get_qemu_binary(params)
+        (support_machine_types, system_versions) = utils_misc.get_support_machine_type(qemu_binary)
+
+        for m_type in support_machine_types:
+            #Using Q35 machine which is not yet fully tested on avocado-vt,so just skip it now
+            if m_type == "q35":
+                break
+            params["machine_type"] = m_type
+            params["start_vm"] = "yes"
+
+            pattern_version1 = "rhel6"
+            pattern_version2 = "rhel7"
+            pattern_version3 = "pc"
+            # pattern_version4 = "q35"
+            match_rhel6 = re.search(pattern_version1, m_type)
+            match_rhel7 = re.search(pattern_version2, m_type)
+            match_pc = re.search(pattern_version3, m_type)
+            # match_q35 = re.search(pattern_version4,m_type)
+
+            pattern = ""
+            if match_rhel6:
+                pattern += "bios.bin"
+            elif match_rhel7 or match_pc:
+                pattern += "bios-256k.bin"
+            else:
+                err_msg = "New machine type needs to check:   is %s " % m_type
+                test.error(err_msg)
```

Comment:
you can use a dictionary biso_bin_dic = {rhel6: bios.bin, rhel7:bios-256k.bin}, then all these lines can be deleted

Reply:
@suqinhuang Thanks, It is very good idea.  I use dictionary to replace these lines.